waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -1479,6 +1478,20 @@
                                 :volumes [{:name "user-home"
                                            :emptyDir {}}]
                                 :terminationGracePeriodSeconds total-sigkill-delay-secs}}}}
+      ;; optional pod anti-affinity to avoid scheduling multiple replicas
+      ;; of the same waiter service on the same kubernetes node
+      ;; https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity
+      (some? pod-anti-affinity)
+      (assoc-in [:spec :template :spec :affinity :podAntiAffinity]
+                (let [affinity-type (case pod-anti-affinity
+                                      :preferred :preferredDuringSchedulingIgnoredDuringExecution
+                                      :required :requiredDuringSchedulingIgnoredDuringExecution
+                                      (throw (ex-info "misconfigured pod-anti-affinity" {:context context})))]
```

Comment:
I agree that ideally it would — but this is part of the replicaset builder context (not the scheduler context), and it's not really part of the scheduler's contract to assume specific values for this. It's part of the given replicaset-builder-fn's contract.

Also, the context provided to the replicaset-builder is semi-dynamic for some reason, which means we'd have to re-validate here anyway. I don't remember why we did this, but we merge values into the context when calling the builder fn:
https://github.com/twosigma/waiter/blob/365fa67f968424a69c58ef8eeb2cdd8d810f6ba5/waiter/src/waiter/scheduler/kubernetes.clj#L2100-L2103